### PR TITLE
Driver framework 2

### PR DESCRIPTION
**Pull request by mcharleb**
New branch for the co-ordinated development for driver_framework integration. It is rebased on master. This is not yet complete and should not yet be merged but the PR allows the changes to be visible.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Where are all the commits I have in this other branch? Is there a particular issue with working off that branch / PR?
https://github.com/PX4/Firmware/pull/3172

**Comment by LorenzMeier** [comment]
The diff here will fail with address sanitizer, as it casts values like "12" to pointers. That's fixed in driver_framework. Ensuring that the critical tools like lldb (no signals) and address sanitizer (no non-address values casted to pointers) work should be our first priority in the rewrite.

File: src/platforms/posix/drivers/gyrosim/gyrosim.cpp [hunk 1]
```diff
@@ -789,15 +753,15 @@ GYROSIM::ioctl(device::file_t *filp, int cmd, unsigned long arg)
 
 			/* set default/max polling rate */
 			case SENSOR_POLLRATE_MAX:
-				return ioctl(filp, SENSORIOCSPOLLRATE, 1000);
+				return devIOCTL(SENSORIOCSPOLLRATE, (void *)1000);
```
Comment (LorenzMeier) [inline src/platforms/posix/drivers/gyrosim/gyrosim.cpp]: Fails with address sanitizer

File: src/platforms/posix/drivers/gyrosim/gyrosim.cpp [hunk 2]
```diff
@@ -789,15 +753,15 @@ GYROSIM::ioctl(device::file_t *filp, int cmd, unsigned long arg)
...
 
 			case SENSOR_POLLRATE_DEFAULT:
-				return ioctl(filp, SENSORIOCSPOLLRATE, GYROSIM_ACCEL_DEFAULT_RATE);
+				return devIOCTL(SENSORIOCSPOLLRATE, (void *)GYROSIM_ACCEL_DEFAULT_RATE);
```
Comment (LorenzMeier) [inline src/platforms/posix/drivers/gyrosim/gyrosim.cpp]: Fails with address sanitizer.

**Comment by mcharleb** [comment]
OK, I have switched over to PX4:driver_franework

**Pull request closed without merging**